packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/pipeline-stages.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import {
+  Pipeline,
+  Stage,
+  Description,
+  Link,
+  css,
+  cx,
+  spacing,
+  Button,
+  Icon,
+} from '@mongodb-js/compass-components';
+
+import type { RootState } from '../../../modules';
+import { stageAdded } from '../../../modules/pipeline';
+import { changeWorkspace } from '../../../modules/workspace';
+
+const containerStyles = css({
+  display: 'grid',
+  gap: spacing[2],
+});
+
+const pipelineContainerStyles = css({
+  gridTemplateAreas: '"pipeline edit"',
+});
+
+const addStageStyles = css({
+  border: 'none',
+  backgroundColor: 'transparent',
+  span: {
+    // LG style is applied first
+    backgroundImage: 'none !important',
+  },
+});
+
+const pipelineStyles = css({
+  gridArea: 'pipeline',
+});
+
+const editButtonStyles = css({
+  gridArea: 'edit',
+});
+
+// todo: remove this post removal of global styles
+const resetParaStyles = css({
+  margin: 'inherit !important',
+});
+
+const PipelineStages: React.FunctionComponent<PipelineStagesProps> = ({
+  isEditing,
+  stages,
+  onStageAdded,
+  onChangeWorkspace,
+}) => {
+  if (stages.filter(Boolean).length === 0) {
```

Comment:
So now that we filter out empty stages I do see this message when opening a new aggregation with a single empty stage:

<img width="880" alt="image" src="https://user-images.githubusercontent.com/5036933/159236791-91731602-9990-42e1-9d3b-19013932dcdb.png">

I intuitively clicked the button to add a new stage because the message said so and nothing changed (because it's another empty stage) and this made me really confused at first, I thought something broke, clicked it a few more times, and only then noticed that I added like 10 more empty stages below the existing empty one.

I wonder if we should think through this behavior here a bit more and sync with Max (or Claudia when she's back)

Reply:
I had a chat about this with @mcasimir and this seems fine. However, maybe we can add an extra if to validate this use case and then show corresponding message?